experimental/traffic-portal/src/app/shared/tp-header/tp-header.component.html

```diff
@@ -19,30 +19,36 @@ <h1>{{title ? title : 'Welcome to Traffic Portal!'}}</h1>
 	<div></div>
 	<nav id="expanded">
 		<ul>
-			<li><a mat-button routerLink="/core/">Home</a></li>
-			<li *ngIf="hasPermission('USER:READ')"><a mat-button routerLink="/core/users">Users</a></li>
-			<li *ngIf="hasPermission('SERVER:READ')"><a mat-button routerLink="/core/servers">Servers</a></li>
+			<li *ngFor="let nav of horizNavs">
+				<a *ngIf="navShown(nav, 'anchor')" mat-button [routerLink]="nav.routerLink">{{nav.text}}</a>
+				<button *ngIf="navShown(nav, 'button')" mat-button (click)="nav?.click()">{{nav.text}}</button>
+			</li>
 			<li>
 				<button mat-icon-button [matMenuTriggerFor]="expandedMenu">
 					<mat-icon>manage_accounts</mat-icon>
 				</button>
 				<mat-menu #expandedMenu="matMenu">
-					<a mat-menu-item routerLink="/core/me">Profile</a>
-					<button mat-menu-item (click)="logout()">Logout</button>
 					<button mat-menu-item [matMenuTriggerFor]="themeMenu">Theme</button>
+					<div *ngFor="let nav of vertNavs">
+						<a *ngIf="navShown(nav, 'anchor')" mat-menu-item [routerLink]="nav.routerLink">{{nav.text}}</a>
+						<button *ngIf="navShown(nav, 'button')" mat-menu-item (click)="nav?.click()">{{nav.text}}</button>
```

Comment:
I'm pretty sure there's a way to do the typing so that button nav items aren't allowed to have a null/undefined `click` property. It seems really similar to what I did for the generic table's context menu items.